source/lac/scalapack.cc

```diff
@@ -914,19 +914,34 @@ NumberType ScaLAPACKMatrix<NumberType>::norm(const char type) const
 
 
 template <typename NumberType>
-void ScaLAPACKMatrix<NumberType>::save(const char *filename) const
+void ScaLAPACKMatrix<NumberType>::save(const char *filename,
+                                       const std::pair<unsigned int,unsigned int> &chunk_size) const
 {
 #ifndef DEAL_II_WITH_HDF5
   (void)filename;
+  (void)chunk_size;
   AssertThrow(false, ExcMessage ("HDF5 support is disabled."));
 #else
+
+  Assert((chunk_size.first <= (unsigned int)n_rows) && (chunk_size.first>0),ExcIndexRange(chunk_size.first,1,n_rows+1));
+  Assert((chunk_size.second <= (unsigned int)n_columns) && (chunk_size.second>0),ExcIndexRange(chunk_size.second,1,n_columns+1));
+
+  std::pair<unsigned int,unsigned int> chunks_size_ = chunk_size;
+  if (chunks_size_.first==numbers::invalid_unsigned_int || chunks_size_.second==numbers::invalid_unsigned_int)
+    {
+      //HDF5, which uses row-major ordering, sees matrix actually in transposed format due to column-major ordering in ScaLAPACK
+      //Consequently, a chunk is a column of the matrix
+      chunks_size_.first = 1;
+      chunks_size_.second = n_rows;
```

Comment:
should not that be `n_columns`?

Reply:
Due to column/row major for Fortran and C it is switched here!
For HDF5 it appears we want to save rows in chunks, which are for ScaLAPACK the columns.

Reply:
i would say you need to use it according to this change somewhere in HDF5, because here those are just defaults. And the description of the function states that first `unsigned int` is used to split rows and the second -- columns. IMO something is inconsistent.

Reply:
(maybe adjust the test to work with non-square matrices?)